### PR TITLE
feat: add kubernetes clients features

### PR DESCRIPTION
**Pull request by pgressa**

### CODE REVIEW

File: starter-core/src/main/java/io/micronaut/starter/feature/k8s/KubernetesReactorClient.java
Comment (jameskleeh) [inline starter-core/src/main/java/io/micronaut/starter/feature/k8s/KubernetesReactorClient.java:76]: Are the two clients incompatible with eachother? If so this seems like a use case for `OneOfFeature` instead of silently removing the non reactive client

File: starter-core/src/main/java/io/micronaut/starter/feature/k8s/KubernetesReactorClient.java
Comment (jameskleeh) [inline starter-core/src/main/java/io/micronaut/starter/feature/k8s/KubernetesReactorClient.java]: This should probably check if micronaut reactor or micronaut reactor http client is part of the features since this would be redundant in that case

File: starter-core/src/main/java/io/micronaut/starter/feature/k8s/KubernetesRxJava2Client.java
Comment (jameskleeh) [inline starter-core/src/main/java/io/micronaut/starter/feature/k8s/KubernetesRxJava2Client.java]: Same comment as above but for micronaut rxjava2

**Review by jameskleeh — CHANGES_REQUESTED** [review]
Is there no kubernetes-rxjava3-client ?

File: starter-core/src/main/java/io/micronaut/starter/feature/k8s/KubernetesReactorClient.java
Comment (pgressa) [reply starter-core/src/main/java/io/micronaut/starter/feature/k8s/KubernetesReactorClient.java:76]: They are compatible.

**Comment by pgressa** [comment]
@jameskleeh no rx3 yet, but good idea, i'll create an issue

File: starter-core/src/main/java/io/micronaut/starter/feature/k8s/KubernetesReactorClient.java
Comment (pgressa) [reply starter-core/src/main/java/io/micronaut/starter/feature/k8s/KubernetesReactorClient.java]: true, will add

File: starter-core/src/main/java/io/micronaut/starter/feature/k8s/KubernetesReactorClient.java
Comment (pgressa) [reply starter-core/src/main/java/io/micronaut/starter/feature/k8s/KubernetesReactorClient.java:76]: It's just it's not needed to reference kubernetes client, since the dependency is part of the reactive clients

**Comment by graemerocher** [comment]
@jameskleeh I don't think the lack of an rxjava3 client should hold this up

**Review by graemerocher — APPROVED** [review]

**Comment by jameskleeh** [comment]
Good to merge once the build passes

**Comment by pgressa** [comment]
@jameskleeh build passed, except status

**Pull request merged**
